18F/identity-idp
# LG-10091 FSM/SSN - Move Extra Analytic Properties to A Shared Concern

## 🎫 Ticket
[LG-10091](https://cm-jira.usa.gov/browse/LG-10091) FSM/SSN - Move extra analytics properties to a shared concern

## 🛠 Summary of changes
- Move extra analytics properties to a shared concern (so that each of the steps in the non-FSM version can use it)
- Refactor `app/controllers/idv/in_person/verify_info_controller.rb` to use shared concern
- Added extra analytics properties to non-FSM ssn step so it would have same_address_as_id

## 📜 Testing Plan

1. Run through the steps👇 on main branch with `in_person_ssn_info_controller_enabled` set to false in application.yml. This shows the analytic property currently being added to the FSM version on the SSN step. (See Scenario 1 in screenshots)

2. Run through the steps👇 on main branch with `in_person_ssn_info_controller_enabled` set to true in application.yml. This shows the analytic property currently not being added to the non-FSM version on the SSN step. (See Scenario 2 in screenshots)

3. Run through the steps👇 on `gyamada/LG-10091-extra-prop` branch with `in_person_ssn_info_controller_enabled` set to true in application.yml. This shows the analytic property has been added to the non-FSM version on the SSN step. (See Scenario 3 in screenshots)

4. Run through the team Ada's flow on `gyamada/LG-10091-extra-prop`. This shows the analytic property has not been added to the SSN step for the doc auth flow. (See Scenario 4 in screenshots) 


Steps:
1. Set `in_person_ssn_info_controller_enabled` to false in application.yml
2. Open file `events.log` inside log
3. Start server to run locally.
4. Create an account. Confirm email. Set up a password. Set up Authentication method. Change url to /verify to Get started verifying your identity. Upload photos to fail proofing and proceed to Try in person.
5. Fill out forms until you land on url /verify/in_person/verify_info
6. Open up `events.log` and search for `ssn`. View the properties for both `IdV: doc auth ssn visited` and `Idv: doc auth ssn submitted`.  Both should have field `same_address_as_id` and the value should match what your selection was. 

## 👀 Screenshots
<details>
<summary>Scenario 1:</summary>
<img width="1079" alt="Screenshot 2023-07-28 at 10 25 55 AM" src="https://github.com/18F/identity-idp/assets/125507397/2633def1-cb32-4b4b-8fd1-74e49ee8fe73">

</details>

<details>
<summary>Scenario 2:</summary>
<img width="1217" alt="Screenshot 2023-07-28 at 10 30 12 AM" src="https://github.com/18F/identity-idp/assets/125507397/81a1f435-5d43-455b-b86e-a3bd3ce4332a">

</details>


<details>
<summary>Scenario 3:</summary>
<img width="1234" alt="Screenshot 2023-07-26 at 8 55 28 AM" src="https://github.com/18F/identity-idp/assets/125507397/e98cad10-8526-42f0-beda-0cf557f04b08">
</details>

<details>
<summary>Scenario 4:</summary>
<img width="1001" alt="Screenshot 2023-07-28 at 10 18 07 AM" src="https://github.com/18F/identity-idp/assets/125507397/5c48491f-f2ac-4d6b-a2bd-1a9750bd474d">
</details>
